src/main/java/com/conveyal/datatools/manager/extensions/mtc/MtcFeedResource.java

```diff
@@ -88,49 +100,45 @@ public void importFeedsForProject(Project project, String authHeader) throws IOE
             in.close();
 
             String json = response.toString();
-            RtdCarrier[] results = mapper.readValue(json, RtdCarrier[].class);
-            for (int i = 0; i < results.length; i++) {
-                //                    String className = "RtdCarrier";
-                //                    Object car = Class.forName(className).newInstance();
-                RtdCarrier car = results[i];
-                //LOG.info("car id=" + car.AgencyId + " name=" + car.AgencyName);
-
+            RtdCarrier[] carriers = mapper.readValue(json, RtdCarrier[].class);
+            // Iterate over carriers found in response and update properties. Also, create a feed source for any carriers
+            // found in the response that do not correspond to an agency ID found in the external feed source properties.
+            for (int i = 0; i < carriers.length; i++) {
+                RtdCarrier carrier = carriers[i];
                 FeedSource source = null;
 
-                // check if a FeedSource with this AgencyId already exists
+                // Check if a FeedSource with this AgencyId already exists.
                 for (FeedSource existingSource : project.retrieveProjectFeedSources()) {
```

Comment:
This would make a database call for each carrier when they could just be prefetched once. It'd be good to move this to happen before the for loop over the carriers.

Reply:
Oh, you're absolutely right. Thanks for catching.